### ISSUE TITLE
FP: Suspicious assignment (when overriding value in if-block)

### ISSUE THREAD

**Issue by cburschka** [kalessil/phpinspectionsea#1588]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) |
| Language level | PHP 7.4

Current behaviour
--
Assignment to array keys in if-blocks can get unexpectedly flagged as "immediately overridden" even though the value has already been used.

```php
<?php

$array = ['id' => 1];

$array['key'] = 1;
$output1 = $array;

$array['key'] = 1.5; // Not flagged.

$array['key'] = 2; // Flagged as suspicious assignment. (Correct!)
$output2 = $array;

if (true) {
$array['key'] = 3; // Not flagged.
$output3 = $array;
}
else {
}

$array['key'] = 4; // Not flagged.
$output4 = $array;

if (true) {
  $array['key'] = 5; // Not flagged.
  $output5 = $array;
}

$array['key'] = 6;   // Flagged as suspicious assignment. (Incorrect!)
$output6 = $array;
```

Expected behaviour
--

Only the assignment to 2 should be flagged as immediately overriding a prior assignment. The assigment to 5 is "used" (by assigning the resulting array to a different variable) before being overridden by the assigment to 6.

The weird behavior in this bug is that it seems to be specifically limited to assigments in an if-block with no else-block - hence why assignment 4 is not flagged, but assignment 6 is.
